infrastructure.tf

```diff
@@ -426,171 +434,276 @@ resource "oci_core_network_security_group_security_rule" "cluster_lb_nsg_rule_5"
 }
 
+resource "oci_core_network_security_group_security_rule" "cluster_lb_nsg_rule_7" {
+  network_security_group_id = oci_core_network_security_group.cluster_lb_nsg.id
+  protocol                  = "6"
+  direction                 = "INGRESS"
+  source                    = local.anywhere
+  tcp_options {
+    destination_port_range {
+      min = 22624
+      max = 22624
+    }
+  }
+}
+ 
 resource "oci_core_network_security_group" "cluster_controlplane_nsg" {
   compartment_id = var.compartment_ocid
   vcn_id         = oci_core_vcn.openshift_vcn.id
   display_name   = "cluster-controlplane-nsg"
 }
-
+ 
 resource "oci_core_network_security_group_security_rule" "cluster_controlplane_nsg_rule_1" {
   network_security_group_id = oci_core_network_security_group.cluster_controlplane_nsg.id
   direction                 = "EGRESS"
   destination               = local.anywhere
   protocol                  = local.all_protocols
 }
-
+ 
 resource "oci_core_network_security_group_security_rule" "cluster_controlplane_nsg_2" {
   network_security_group_id = oci_core_network_security_group.cluster_controlplane_nsg.id
   protocol                  = local.all_protocols
   direction                 = "INGRESS"
   source                    = var.vcn_cidr
 }
-
+ 
+# new nsg rule for api 
+resource "oci_core_network_security_group_security_rule" "cluster_controlplane_nsg_3" {
+  network_security_group_id = oci_core_network_security_group.cluster_controlplane_nsg.id
+  protocol                  = "6"
+  direction                 = "INGRESS"
+  source                    = local.anywhere
+  tcp_options {
+    destination_port_range {
+      min = 6443
+      max = 6443
+    }
+  }
+}
+ 
 resource "oci_core_network_security_group" "cluster_compute_nsg" {
   compartment_id = var.compartment_ocid
   vcn_id         = oci_core_vcn.openshift_vcn.id
   display_name   = "cluster-compute-nsg"
 }
-
+ 
 resource "oci_core_network_security_group_security_rule" "cluster_compute_nsg_rule_1" {
   network_security_group_id = oci_core_network_security_group.cluster_compute_nsg.id
   direction                 = "EGRESS"
   destination               = local.anywhere
   protocol                  = local.all_protocols
 }
-
+ 
 resource "oci_core_network_security_group_security_rule" "cluster_compute_nsg_2" {
   network_security_group_id = oci_core_network_security_group.cluster_compute_nsg.id
   protocol                  = local.all_protocols
   direction                 = "INGRESS"
   source                    = var.vcn_cidr
 }
 
-resource "oci_network_load_balancer_network_load_balancer" "openshift_lb" {
-  compartment_id             = var.compartment_ocid
-  subnet_id                  = oci_core_subnet.public.id
-  display_name               = "openshift_lb"
-  is_private                 = false
+resource "oci_load_balancer_load_balancer" "openshift_internal_lb" {
+  compartment_id = var.compartment_ocid
+  display_name = "${var.cluster_name}-openshift_internal_lb"
```

Comment:
excellent change, in general on OCP deployments, the cloud resource name is prefixed w/ cluster name (or "infra ID")  🚀

Reply:
The remaining resources which don't have cluster_name in the display name all belong within the VNC which does have the cluster name, so it's clear that they belong